### PR TITLE
update image classification

### PR DESCRIPTION
**Pull request by MorganeLu**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] [local build the book](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-book)
- [ ] review the artifacts from local build
- [ ] [local build the slides](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-slides-optional) if appliable
- [ ] Unit test if appliable

# Checklist:

- [x] My code follows the [style guidelines](https://github.com/open-academy/machine-learning/blob/main/STYLE_GUIDE.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have applied tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md [hunk 1]
```diff
@@ -0,0 +1,1034 @@
...
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.11.5
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+# image classification
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/deep-learning/image-classification.md:16]: Imange

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md [hunk 2]
```diff
@@ -0,0 +1,1034 @@
...
+
+## What is image classification?
+
+In this chapter we will introduce the Image Classification problem, which is the task of assigning an input image one label from a fixed set of categories. This is one of the core problems in Computer Vision that, despite its simplicity, has a large variety of practical applications.
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/deep-learning/image-classification.md:20]: image classification all in lower case?

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md [hunk 3]
```diff
@@ -0,0 +1,1034 @@
...
+
+Let me give you an example. in the image below an image classification model takes a single image and assigns probabilities to 4 labels, {cat, dog, hat, mug}. As shown in the image, keep in mind that to a computer an image is represented as one large 3-dimensional array of numbers. In this example, the cat image is 248 pixels wide, 400 pixels tall, and has three color channels Red,Green,Blue (or RGB for short). Therefore, the image consists of 248 x 400 x 3 numbers, or a total of 297,600 numbers. Each number is an integer that ranges from 0 (black) to 255 (white). Our task is to turn this quarter of a million numbers into a single label, such as “cat”.
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/deep-learning/image-classification.md:22]: In the image below.  Could you install a grammarly plugin in the vscode and fix the grammar issue? Or you can copy paste the text to any grammar correct tool for fix too.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md [hunk 4]
```diff
@@ -0,0 +1,1034 @@
...
+
+:::{figure-md} 01_example_of_cls
+<img src="../../images/deep-learning/imgcls/01_classify_eg.png" width="90%" class="bg-white mb-1">
+
+Example of the image classification task
+:::
+
+The task in Image Classification is to predict a single label (or a distribution over labels as shown here to indicate our confidence) for a given image. Images are 3-dimensional arrays of integers from 0 to 255, of size Width x Height x 3. The 3 represents the three color channels Red, Green, Blue.
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/deep-learning/image-classification.md:30]: Please make the image classification to be consistently capitalized.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md [hunk 5]
```diff
@@ -0,0 +1,1034 @@
...
+
+## Challenges
+
+Since this task of recognizing a visual concept (e.g. cat) is relatively trivial for a human to perform, it is worth considering the challenges involved from the perspective of a Computer Vision algorithm. As we present (an inexhaustive) list of challenges below, keep in mind the raw representation of images as a 3-D array of brightness values:
+- Viewpoint variation. A single instance of an object can be oriented in many ways with respect to the camera,
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/deep-learning/image-classification.md:35]: Use . At the end if you use upper case for each line.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md [hunk 6]
```diff
@@ -0,0 +1,1034 @@
...
+
+### VGGNet
+
+The VGG (Visual Geometry Group) multilayer network model has 19 more layers than AlexNet, verifying that increasing the depth in the network structure can directly affect the model performance. The design idea of VGG is to increase the depth of the network and use a small size convolutional kernel instead. As shown in the figure below, three 3×3 convolutional kernels are used to replace the 7×7 convolutional kernels in AlexNet, and two 3×3 convolutional kernels are used to replace the 5×5 convolutional kernels, which can increase the depth of the network and improve the model effect while ensuring the same perceptual field. The number of model parameters and operations can be reduced by using smaller 3×3 Filters, and the image feature information can be better retained. The specific advantages of the improvement are summarized as follows. 
+- Using small 3×3 filters to replace large convolutional kernels
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/deep-learning/image-classification.md:59]: Add an empty line before this.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md [hunk 7]
```diff
@@ -0,0 +1,1034 @@
...
+The most important part of ResNet should be the residual block, and here is the structure.
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/deep-learning/image-classification.md:310]: Add an empty line before this.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md [hunk 8]
```diff
@@ -0,0 +1,1034 @@
...
+
+# For Win/Mac
+# Download from the offical website
+```
+
+### ImageNet-1000
+
+[ImageNet](https://image-net.org/) is an image dataset organized according to the WordNet hierarchy. Each meaningful concept in WordNet, possibly described by multiple words or word phrases, is called a "synonym set" or "synset". There are more than 100,000 synsets in WordNet; the majority of them are nouns (80,000+). In ImageNet, we aim to provide on average 1000 images to illustrate each synset. Images of each concept are quality-controlled and human-annotated. In its completion, we hope ImageNet will offer tens of millions of cleanly labeled and sorted images for most of the concepts in the WordNet hierarchy.
+
+#### Download
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/deep-learning/image-classification.md:1006]: Use {note} instead?

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md [hunk 9]
```diff
@@ -0,0 +1,1034 @@
...
+
+If you want to download this dataset, please visit [kaggle](https://www.kaggle.com/c/imagenet-object-localization-challenge/overview/description) for more information.
+
+## Standards
+
+### top-1 accuracy
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/deep-learning/image-classification.md:1012]: Top?

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md [hunk 10]
```diff
@@ -0,0 +1,1034 @@
...
+
+If your predicted label takes the largest one inside the final probability vector as the prediction result, and if the one with the highest probability in your prediction result is correctly classified, then the prediction is correct. Otherwise, the prediction is wrong.
+
+### top-5 accuracy
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/deep-learning/image-classification.md:1016]: Top?

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md [hunk 11]
```diff
@@ -0,0 +1,1034 @@
...
+## Classic datasets
+
+As we said before, image classification task is mainly trained by datasets, so the importance of dataset is obvious. Here, we will introduce some widely-used datasets.
+
+### CIFAR-10/100
+
+The [CIFAR-10 dataset](http://www.cs.toronto.edu/~kriz/cifar.html) consists of 60000 32x32 colour images in 10 classes, with 6000 images per class. There are 50000 training images and 10000 test images. The dataset is divided into five training batches and one test batch, each with 10000 images. The test batch contains exactly 1000 randomly-selected images from each class. The training batches contain the remaining images in random order, but some training batches may contain more images from one class than another. Between them, the training batches contain exactly 5000 images from each class.
+The CIFAR-100 dataset is just like the CIFAR-10, except it has 100 classes containing 600 images each. There are 500 training images and 100 testing images per class. The 100 classes in the CIFAR-100 are grouped into 20 superclasses. Each image comes with a "fine" label (the class to which it belongs) and a "coarse" label (the superclass to which it belongs).
+
+#### Download
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/deep-learning/image-classification.md:991]: Use note block.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md [hunk 12]
```diff
@@ -0,0 +1,1034 @@
...
+```
+
+## Classic datasets
+
+As we said before, image classification task is mainly trained by datasets, so the importance of dataset is obvious. Here, we will introduce some widely-used datasets.
+
+### CIFAR-10/100
+
+The [CIFAR-10 dataset](http://www.cs.toronto.edu/~kriz/cifar.html) consists of 60000 32x32 colour images in 10 classes, with 6000 images per class. There are 50000 training images and 10000 test images. The dataset is divided into five training batches and one test batch, each with 10000 images. The test batch contains exactly 1000 randomly-selected images from each class. The training batches contain the remaining images in random order, but some training batches may contain more images from one class than another. Between them, the training batches contain exactly 5000 images from each class.
+The CIFAR-100 dataset is just like the CIFAR-10, except it has 100 classes containing 600 images each. There are 500 training images and 100 testing images per class. The 100 classes in the CIFAR-100 are grouped into 20 superclasses. Each image comes with a "fine" label (the class to which it belongs) and a "coarse" label (the superclass to which it belongs).
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/deep-learning/image-classification.md:989]: Give a new line.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md [hunk 13]
```diff
@@ -0,0 +1,1034 @@
...
+        x = Linear(x)
+
+        # x = tf.reshape(x, [-1, 10])
+        return x
+```
+
+### MobileNet
+
+MobileNets are based on a streamlined architecture that uses depth-wise separable convolutions to build light weight deep neural networks. We introduce two simple global hyper-parameters that efficiently trade off between latency and accuracy. These hyper-parameters allow the model builder to choose the right sized model for their application based on the constraints of the problem.
+Besides, the standard convolutional filters for normal CNNs are replaced by two layers: depthwise convolution and pointwise convolution to build a depthwise separable filter.
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/deep-learning/image-classification.md:675]: Give a new line. If you are using vscode, please install a markdownlint plugin.

**Review by zhangqi444 — APPROVED** [review]

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md (hunk 1 above)
Comment (reveurmichael) [reply open-machine-learning-jupyter-book/deep-learning/image-classification.md:16]: Image

Comment (MorganeLu) [reply open-machine-learning-jupyter-book/deep-learning/image-classification.md]: Done.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md (hunk 2 above)
Comment (MorganeLu) [reply open-machine-learning-jupyter-book/deep-learning/image-classification.md]: Done.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md (hunk 3 above)
Comment (MorganeLu) [reply open-machine-learning-jupyter-book/deep-learning/image-classification.md]: This problem has done. I use the pycharm since I am not familiar with vscode, but I will find the similar package for pycharm.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md (hunk 4 above)
Comment (MorganeLu) [reply open-machine-learning-jupyter-book/deep-learning/image-classification.md]: Done.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md (hunk 5 above)
Comment (MorganeLu) [reply open-machine-learning-jupyter-book/deep-learning/image-classification.md]: Done.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md (hunk 6 above)
Comment (MorganeLu) [reply open-machine-learning-jupyter-book/deep-learning/image-classification.md]: Done.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md (hunk 7 above)
Comment (MorganeLu) [reply open-machine-learning-jupyter-book/deep-learning/image-classification.md:312]: Done.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md (hunk 8 above)
Comment (MorganeLu) [reply open-machine-learning-jupyter-book/deep-learning/image-classification.md:1014]: Yes, it has been changed.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md (hunk 9 above)
Comment (MorganeLu) [reply open-machine-learning-jupyter-book/deep-learning/image-classification.md]: Done.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md (hunk 10 above)
Comment (MorganeLu) [reply open-machine-learning-jupyter-book/deep-learning/image-classification.md]: Done.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md (hunk 11 above)
Comment (MorganeLu) [reply open-machine-learning-jupyter-book/deep-learning/image-classification.md:997]: Done.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md (hunk 13 above)
Comment (MorganeLu) [reply open-machine-learning-jupyter-book/deep-learning/image-classification.md:678]: Done, and I will find a substitution for pycharm.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md (hunk 3 above)
Comment (zhangqi444) [reply open-machine-learning-jupyter-book/deep-learning/image-classification.md]: Sure.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md (hunk 8 above)
Comment (zhangqi444) [reply open-machine-learning-jupyter-book/deep-learning/image-classification.md:1014]: I mean use {note} for the `#### Download` It is not necessary to have a 4th level title.

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md (hunk 11 above)
Comment (zhangqi444) [reply open-machine-learning-jupyter-book/deep-learning/image-classification.md:997]: same here

**Review by zhangqi444 — CHANGES_REQUESTED** [review]

File: open-machine-learning-jupyter-book/deep-learning/image-classification.md (hunk 11 above)
Comment (MorganeLu) [reply open-machine-learning-jupyter-book/deep-learning/image-classification.md]: Done.

**Review by zhangqi444 — APPROVED** [review]

**Review by zhangqi444 — APPROVED** [review]
